pkg/handlers/session.go

```diff
@@ -75,12 +76,23 @@ func requireValidSession(kotsStore store.Store, w http.ResponseWriter, r *http.R
 	}
 
 	if time.Now().After(sess.ExpiresAt) {
+		if err := kotsStore.DeleteSession(sess.ID); err != nil {
+			logger.Error(errors.Wrapf(err, "session expired. failed to delete expired session %s", sess.ID))
+		}
 		err := errors.New("session expired")
 		response := ErrorResponse{Error: err.Error()}
 		JSON(w, http.StatusUnauthorized, response)
 		return nil, err
 	}
 
+	// extend the session by an hour, if user is still active after an hour
+	if delta := time.Until(sess.ExpiresAt); delta.Hours() <= (SessionTimeout - 1*time.Hour).Hours() {
+		sess.ExpiresAt = sess.ExpiresAt.Add(time.Hour * 1)
+		if err := kotsStore.UpdateSessionExpiresAt(sess.ID, sess.ExpiresAt); err != nil {
+			logger.Error(errors.Wrapf(err, "failed to update session expiry %s", sess.ID))
+		}
+	}
```

Comment:
```suggestion
	// give the user the full session timeout if they have been active at least an hour
	if time.Now().Add(SessionTimeout - time.Hour).After(sess.ExpiresAt){
		sess.ExpiresAt = time.Now().Add(SessionTimeout)
		if err := kotsStore.UpdateSessionExpiresAt(sess.ID, sess.ExpiresAt); err != nil {
			logger.Error(errors.Wrapf(err, "failed to update session expiry %s", sess.ID))
		}
	}
```

Given 5 requests in a row from a user with 1 hour remaining, your code would update it to 2 then 3 then 4 then 5 then 6 hours remaining - just setting it to a constant value is both easier to read and more efficient.